core/testing/include/Kiso_Testing.h

```diff
@@ -0,0 +1,192 @@
+ *                  codes. In case of error, one can refer to this enumeration to identify which source file has issued
+ *                  it. 
+*/
+enum KISO_TESTING_Modules_E
+{
+    KISO_MODULE_ID_TESTING = 1,
+    KISO_MODULE_ID_TESTING_TESTREGISTRY,         /**< KISO_TESTS_MODULE_ID_TESTREGISTRY */
+    KISO_MODULE_ID_TESTING_CCHANNEL,             /**< KISO_TESTS_MODULE_ID_CCHANNEL */
+    KISO_MODULE_ID_TESTING_SERIALMSGTRANSCEIVER, /**< KISO_TESTS_MODULE_ID_SERIALMSGTRANSCEIVER */
+    KISO_MODULE_ID_TESTING_TESTRUNNER,           /**< KISO_TESTS_MODULE_ID_TESTRUNNER */
+    KISO_MODULE_ID_TESTING_SERIAL,               /**< KISO_TESTS_MODULE_ID_SERIAL */
+    KISO_MODULE_ID_TESTING_TESTENTRY,
+};
+
+/**
+ * @brief           Enumerates the special return codes that could be returned from the different units in the 
+ *                  Kiso_Testing library.
+*/
+enum KISO_TESTING_Retcodes_E
+{
+    TEST_RETCODE_TEST_SUITE_ALREADY_REGISTERED = RETCODE_FIRST_CUSTOM_CODE,
+    TEST_RETCODE_TEST_CASE_ALREADY_REGISTERED,
+    TEST_RETCODE_CCHANNEL_INITIALISATION_FAILED,
+    TEST_RETCODE_INCOMPLETE_MESSAGE_RECEIVED,
+    TEST_RETCODE_CRC_MISMATCH,
+    TEST_RETCODE_VERSION_MISMATCH,
+    TEST_RETCODE_TLVELEMENT_NOT_FOUND,
+    TEST_RETCODE_REPORT_TIMEOUT,
+    TEST_RETCODE_CCHANNEL_NOT_SPECIFIED,
+};
+
+/**
+ * @brief           Enumerates the types of messages exchanged between the Test_Executor and the Test_Controller 
+ */
+enum CCMsg_MessageType_E
+{
+    CCMSG_TYPE_COMMAND,
+    CCMSG_TYPE_REPORT,
+    CCMSG_TYPE_ACK,
+};
+
+/**
+ * @brief           Encapsulates the elements composing the message header.
+ */
+typedef struct MessageHeader_S
+{
+    uint8_t messageInfo; /**< version is 2 bits, message type is 2 bits and the remaining 4 bits are reserved */
+    uint8_t messageToken;
+    uint8_t messageType;
+    uint8_t errorCode;
+    uint8_t testEntry;
+    uint8_t testSuite;
+    uint8_t testCase;
+    uint8_t payloadLength;
+} MsgHdr_T;
+
+/**
+ * @brief           Encapsulates the elements composing a TLV object(type, length and value)
+ */
+typedef struct TlvElement_S
+{
+    uint8_t type;
+    uint8_t length;
+    char *value;
+} TlvElt_T;
+
+/**
+ *  @brief          Encapsulates the elements composing a message..
+ */
+typedef struct CCMsg_S
+{
+    MsgHdr_T header; /**< Header of the message */
+    uint8_t payload[TEST_MAX_PAYLOAD_LENGTH];
+    TlvElt_T tlvArray[CCHANNEL_MAX_NUMBER_OF_TLV_ELEMENTS]; /**< Parsed TLV array (from receive buffer) */
+    uint8_t payloadIndex;
+    uint8_t tlvArrayIndex;
+    uint8_t numberOfTlvElements; /**< Number of TLV elements for the message */
+    uint8_t rebootCounter;
+    bool isFree : 1; /**< This bit indicates whether the received message is processed by test runner or not */
+} CCMsg_T;
+
+/**
+ * @brief           Defines a prototype type for the setup functions of the test suites and test cases.
+ */
+typedef Retcode_T (*StpFct_T)(CCMsg_T *ccmsg);
```

Comment:
we need to talk about the naming conventions.. First rule in my opinion it should be pronounceable and clear.
here it could be StopFunction or a StepFunction.  well established abbreviations are ok.